crates/notedeck_ui/src/note/video.rs

```diff
@@ -0,0 +1,400 @@
+use egui::{
+    vec2, Align, Align2, Area, Color32, FontId, Key, Layout, Order, Pos2, Rect, Response, RichText,
+    Sense, Ui, UiBuilder, WidgetText,
+};
+use egui_video::AudioDevice;
+use egui_video::{ensure_initialized, Player, PlayerState};
+use hex::encode;
+use poll_promise::Promise;
+use sdl2;
+use sha2::{Digest, Sha256};
+use std::{
+    cell::RefCell,
+    collections::HashMap,
+    env,
+    fs::{self, File},
+    io::copy,
+    path::PathBuf,
+};
+use ureq;
+
+enum VideoSlot {
+    Loading {
+        promise: Promise<Result<String, String>>,
+    },
+    Ready {
+        player: Player,
+        started: bool,
+    },
+    Failed(String),
+}
+
+enum VideoStatus {
+    Loading,
+    Error(String),
+}
+
+thread_local! {
+    static VIDEO_PLAYERS: RefCell<HashMap<String, VideoSlot>> = RefCell::new(HashMap::new());
+    static AUDIO_SUPPORT: RefCell<Option<AudioSupport>> = RefCell::new(None);
+    static FULLSCREEN_VIDEOS: RefCell<HashMap<String, bool>> = RefCell::new(HashMap::new());
+}
```

Comment:
global variables are not allowed in this codebase, even if they are thread local. state should be managed in an struct that is passed in as reference.